cibuildwheel/options.py

```diff
@@ -150,25 +152,78 @@ class ConfigOptionError(KeyError):
     pass
 
 
-def _dig_first(*pairs: tuple[Mapping[str, Setting], str], ignore_empty: bool = False) -> Setting:
+class Inherit(enum.Enum):
+    NONE = enum.auto()
+    APPEND = enum.auto()
+    PREPEND = enum.auto()
+
+
+def _dig_first(
+    *pairs: tuple[Mapping[str, Setting], str, Inherit], ignore_empty: bool = False
+) -> Setting:
     """
-    Return the first dict item that matches from pairs of dicts and keys.
-    Will throw a KeyError if missing.
+    Return the dict items that match from pairs of dicts and keys. The third
+    value (enum) indicates if values should merge.
+
+    The following idiom can be used to get the first matching value:
 
-    _dig_first((dict1, "key1"), (dict2, "key2"), ...)
+        _dig_first((dict1, "key1", Inherit.NONE), (dict2, "key2", Inherit.NONE), ...)))
     """
     if not pairs:
         msg = "pairs cannot be empty"
         raise ValueError(msg)
 
-    for dict_like, key in pairs:
+    commands: list[tuple[Setting, Inherit]] = []
+
+    for dict_like, key, merge in pairs:
         if key in dict_like:
             value = dict_like[key]
-
             if ignore_empty and value == "":
                 continue
 
-            return value
+            commands.append((value, merge))
+            if merge == Inherit.NONE:
+                break
+
+    if len(commands) == 1:
+        return commands[0][0]
+
+    if len(commands) > 1:
+        if isinstance(commands[0][0], list):
+            retlist: list[str] = []
+            for c in commands:
+                if c[1] == Inherit.PREPEND:
+                    assert isinstance(c[0], list)
+                    retlist.extend(c[0])
+            for c in commands:
+                if c[1] == Inherit.NONE:
+                    assert isinstance(c[0], list)
+                    retlist.extend(c[0])
+            for c in commands[::-1]:
+                if c[1] == Inherit.APPEND:
+                    assert isinstance(c[0], list)
+                    retlist.extend(c[0])
```

Comment:
I'm not really following what's happening here... maybe it's a little too clever/tricky of a solution?